crates/rpc/rpc-builder/src/auth.rs

```diff
@@ -48,17 +54,30 @@ impl AuthServerConfig {
     }
 
     /// Convenience function to start a server in one step.
-    pub async fn start(self, module: AuthRpcModule) -> Result<AuthServerHandle, RpcError> {
-        let Self { socket_addr, secret, server_config, ipc_server_config, ipc_endpoint } = self;
+    pub async fn start(self, module: AuthRpcModule) -> Result<AuthServerHandle, RpcError>
+    where
+        Middleware: RethRpcMiddleware,
```

Comment:
for this usecase we need to install this as the `tower::ServiceBuilder::new()` middleware instead